Makefile.toml

```diff
@@ -261,21 +261,33 @@ cargo install \
 '''
 ]
 
-[tasks.build-packages]
+# Builds a package including its build-time and runtime dependency packages.
+[tasks.build-package]
 dependencies = ["build-tools", "publish-setup"]
+script_runner = "bash"
 script = [
 '''
+set -e
+if [ -z "${PACKAGE}" ]; then
+    echo "the PACKAGE environment variable must be set. for example:"
+    echo "cargo make build-package -e PACKAGE=kernel"
+    exit 1
+fi
+
 export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
 cargo build \
   ${CARGO_BUILD_ARGS} \
   ${CARGO_MAKE_CARGO_ARGS} \
-  --manifest-path packages/Cargo.toml \
-  --all
+  --manifest-path "packages/${PACKAGE}/Cargo.toml"
 '''
 ]
 
+[tasks.build-kernel]
+env = { "PACKAGE" = "kernel" }
+run_task = "build-package"
+
 [tasks.build-kmod-kit]
-dependencies = ["build-packages"]
+dependencies = ["build-kernel"]
```

Comment:
This is a bit of a mess and it's my fault.

When we introduce the 5.10 kernel, we'll need two of these kmod kits:
* one that has the toolchain + 5.4 kernel files
* another that has the toolchain + 5.10 kernel files

In this new approach, we'd either need a way to express a dependency on something cargo should build that is not a package or an image, or else we'd need to create the kmod kit as a side effect of building the variant - like we do today with migrations.

Maybe a top-level `archives` directory, an ArchiveBuilder component in `buildsys`, where we expect an `archives/{name}/{name}` script?

I didn't use the `rpm2migrations` approach for the kmod kit because I think it's ugly, and also because it requires network access and `buildsys` invoked stuff is forbidden to touch the network.

There are three related use cases, all of which involve post-processing RPMs:
* find the kernel-devel RPM, unpack it, repackage its contents
* find the migrations RPM and do the same thing
* find all debuginfo RPMs and extract the symbols

The debuginfo one has the potential to be pretty slow; we might implement that as a "side effect" for building a package so that we don't batch it all up at the end.

That would incline me toward the "side effect" treatment of the kmod kit also.

Reply:
Should we create an issue for this? I think I would like to *not* build the kmod-kits and debug symbols most of the time. I agree that `rpm2migrations` is not something we want to emulate for future 'archives' that don't go into the Bottlerocket image.